garrysmod/lua/includes/modules/outline.lua

```diff
@@ -0,0 +1,117 @@
+
+module("outline",package.seeall)
+
+local List = {}
+local RenderEnt = NULL
+
+local CopyMat		= Material("pp/copy")
+local OutlineMat	= CreateMaterial("OutlineMat","UnlitGeneric",{["$ignorez"] = 1,["$alphatest"] = 1})
+local StoreTexture	= render.GetScreenEffectTexture(0)
+local DrawTexture	= render.GetScreenEffectTexture(1)
+
+--[[-------------------------------
+	Modes:
+	0 - Always draw
+	1 - Draw if not visible
+	2 - Draw if visible
+--]]-------------------------------
+
+function Add(ents,color,mode)
+	if !istable(ents) then ents = {ents} end	--Support for passing Entity as first argument
+	if table.IsEmpty(ents) then return end		--Do not pass empty tables
```

Comment:
table.IsEmpty returns false for tables like ``{foo = 5}`` but you only use numerical keys for iteration. You should check if ents[1] is not nil, instead.

Reply:
that method used in halo module, i copy that.

Reply:
That is a mistake then, I will fix that.

Reply:
Actually, halos use ``pairs`` on the Ents table so it iterates over non-numerical keys. You don't do that here, you use ``ipairs``